File: Development/WoodLotFrontEnd/src/app/entita/prodottoOrdine/prodotto-ordine.ts
import {Contadino} from "../contadino/contadino";
import {UsoLocale} from "../usoLocale/uso-locale";
import {Beneficio} from "../beneficio/beneficio";
import {Paese} from "../paese/paese";

export class ProdottoOrdine {
  public id: number
  public prezzoUnitario: number
  public stato: String
  public contadino: Contadino | undefined
  public descrizione: String | undefined
  public frutta: number | undefined
  public nomeAlbero: String
  public usiLocali: Array<UsoLocale>
  public benefici: Array<Beneficio>
  public dataAssegnazione: Date | undefined
  public dataModifica: Date | undefined
  public paeseOrigine: Paese

  constructor(id: number, prezzo: number,
              data: Date, stato: String, nomeAlbero: String, usiLocali: Array<UsoLocale>, benefici: Array<Beneficio>, paeseOrigine: Paese,
              contadino?: Contadino, descrizione?: String, frutta?: number, dataAssegnazione?: Date,
              dataModifica?: Date) {
    this.id = id
    this.dataAssegnazione = dataAssegnazione
    this.stato = stato
    this.prezzoUnitario = prezzo
    this.contadino = contadino
    this.descrizione = descrizione
    this.frutta = frutta
    this.benefici = benefici
    this.nomeAlbero = nomeAlbero
    this.usiLocali = usiLocali
    this.dataModifica = dataModifica
    this.paeseOrigine = paeseOrigine
  }
}

File: Development/WoodLotBackEnd/src/main/java/it/unisa/WoodLot/model/entity/ProdottoOrdine.java
package it.unisa.WoodLot.model.entity;


import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.List;

/**
 * Classe che modella un prodotto di un ordine
 *
 * @author Alessia Ture
 */
@Data
@Entity
@Table(name = "prodotto_ordine")
public class ProdottoOrdine implements Serializable {
    public enum Stato {
        Assegnato,
        NonAssegnato,
        Piantato,
        Bocciolo,
        Fiore,
        Frutto,
        Dormienza,
        Senescenza,
        Revisione,
        Riassegnazione
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "prezzo_unitario", nullable = false)
    private double prezzoUnitario;

    @OneToOne()
    @JoinColumn(name = "paese_origine", referencedColumnName = "nome")
    private PaeseOrigine paeseOrigine;

    @Column(name = "stato", nullable = false)
    private Stato stato;

    @Column(name = "data_assegnazione")
    private Date dataAssegnazione;

    @Column(name = "data_modifica")
    private Date dataModifica;

    @ManyToOne(cascade = {CascadeType.ALL})
    @JoinColumn(name = "contadino")
    private Contadino contadino;

    @Column(name = "descrizione")
    private String descrizione;

    @Column(name = "frutta")
    private Double frutta;

    @Column(name = "CO2", nullable = false)
    private int anidrideCarbonicaAssorbita;

    @Column(name = "nome_albero")
    private String nomeAlbero;

    @ManyToMany
    @JoinColumn(name = "uso_locale", referencedColumnName = "nome")
    private List<UsoLocale> usiLocali;

    @ManyToMany
    @JoinColumn(name = "beneficio", referencedColumnName = "nome")
    private List<Beneficio> benefici;

    public ProdottoOrdine() {
        frutta = (double) 0;
    }

    public double getFrutta() {
        if (frutta == null)
            return 0;
        else return frutta;
    }
}
